scaling_performance/cluster_limits.adoc

```diff
@@ -0,0 +1,118 @@
+[[scaling-performance-cluster-limits]]
+= Cluster Limits
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+This topic summarizes the limits for objects in {product-title}.
+
+In most cases, exceeding these thresholds results in lower overall performance.
+It does not necessarily mean that the cluster will fail over.
+
+Some of the limits represented in this topic are given for the largest possible
+cluster. For smaller clusters, the limits are proportionally lower.
+
+There are many factors that influence the stated thresholds, including the etcd
+version or storage data format.
+
+[[scaling-performance-current-cluster-limits]]
+== Current Cluster Limits
+
+[options="header",cols="3,2,2"]
+|===
+| Limit Type |Current Limit| Long-term Goal
+
+|Total number of all objects footnoteref:[allobjects,There is a limitation on the total number of objects on the system, as this affects etcd and its resource consumption.]
+| 250,000
+| 1,000,000
+
+| Number of nodes footnoteref:[numberofnodes,Having clusters with more than 5000 nodes is not the best option. Consider splitting into multiple clusters.]
+| 5,000
+| 5,000
+
+| Number of pods
+| 150,000
+| 500,000
+
+| Number of pods per node footnoteref:[podspernodeorcore, The limit for number of pods on a given node is the minimum from the _pod per node_ and _pods per core multiplied by the number of cores of a node_.]
+| 100
+| 100
+
+| Number of pods per core footnoteref:[podspernodeorcore]
+| 10
+| 10
+
+| Number of namespaces
+| 10,000
+| 100,000
+
+| Number of pods per namespace footnoteref:[objectpernamespace,There are
+a number of control loops in the system that need to iterate over all objects
+in a given namespace as a reaction to some changes in state. Having a large
+number of objects of a given type in a single namespace can make those loops
+expensive and slow down processing given state changes.]
+| 15,000
+| 50,000
+
+| Number of services footnoteref:[servicesandendpoints,Each service port and each service back-end has a corresponding entry in iptables. The number of back-ends of a given service impact the size of the endpoints objects, which impacts the size of data that is being sent all over the system.]
+| 10,000
+| 100,000
+
+| Number of all services back-ends
+| To be determined
+| 500,000
```

Comment:
Please remove all rows that contain "To be determined".  I think we should handle these as research projects and do those as one-offs as more specific requirements are handed down.

Reply:
OK, please update accordingly, and I'll take another pass at review.  Thanks!